File: 55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size()-1;
        while (n!=0) {
            int tmp = n; // var tmp to check if n is changed or not
            for(int i=0; i<n; i++){
                if(nums[i] >= n-i) n = i;
            }
            if(tmp == n) break;
        }

        return (!n) ? true : false; 
    }
};

File: 55-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length-1;
        while(n != 0){
            int tmp = n;
            for(int i=0; i<n; i++){
                if(nums[i] >= n-i) n = i;
            }
            if(n == tmp) break;
        }
        return (n == 0) ? true : false;
    }
}
